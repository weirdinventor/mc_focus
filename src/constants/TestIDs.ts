export const TestIDs = {
  tests: {
    CTextInputBottomText: 'C_TEXT_INPUT_BOTTOM_TEXT',
    Screen: {
      SafeAreaView: 'SCREEN_SAFE_AREA_VIEW',
      AnimatedView: 'SCREEN_ANIMATED_VIEW',
    },
    Wrapper: {
      SafeAreaView: 'WRAPPER_SAFE_AREA_VIEW',
      AnimatedView: 'WRAPPER_ANIMATED_VIEW',
    },
  },
};
