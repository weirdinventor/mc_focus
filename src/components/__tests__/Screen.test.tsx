import React from 'react';
import { render } from '@testing-library/react-native';
import { Screen } from '@components/Screen';
import { TestIDs } from '@constants/TestIDs';

jest.mock('@react-navigation/elements', () => ({
  useHeaderHeight: jest.fn().mockReturnValue(50),
}));

describe('[COMPONENTS]: Screen', () => {
  it('renders correctly', () => {
    const tree = render(<Screen />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('correctly sets default edges', () => {
    const { getByTestId } = render(<Screen />);
    const safeAreaView = getByTestId(TestIDs.tests.Screen.SafeAreaView);
    const edges = {
      bottom: 'additive',
      left: 'off',
      right: 'off',
      top: 'additive',
    };

    expect(safeAreaView.props.edges).toEqual(edges);
  });

  it('correctly sets edges with fullscreen', () => {
    const { getByTestId } = render(<Screen fullscreen />);
    const safeAreaView = getByTestId(TestIDs.tests.Screen.SafeAreaView);
    const edges = {
      bottom: 'off',
      left: 'off',
      right: 'off',
      top: 'off',
    };

    expect(safeAreaView.props.edges).toEqual(edges);
  });

  it('correctly sets edges with withoutTopEdge', () => {
    const { getByTestId } = render(<Screen withoutTopEdge />);
    const safeAreaView = getByTestId(TestIDs.tests.Screen.SafeAreaView);
    const edges = {
      bottom: 'additive',
      left: 'off',
      right: 'off',
      top: 'off',
    };

    expect(safeAreaView.props.edges).toEqual(edges);
  });

  it('correctly sets edges with withoutBottomEdge', () => {
    const { getByTestId } = render(<Screen withoutBottomEdge />);
    const safeAreaView = getByTestId(TestIDs.tests.Screen.SafeAreaView);
    const edges = {
      bottom: 'off',
      left: 'off',
      right: 'off',
      top: 'additive',
    };

    expect(safeAreaView.props.edges).toEqual(edges);
  });

  it('correctly handles default padding', () => {
    const { getByTestId } = render(<Screen />);
    const AnimatedView = getByTestId(TestIDs.tests.Screen.AnimatedView);

    expect(AnimatedView).toHaveStyle({ paddingHorizontal: 24 });
  });

  it('correctly handles padding with noHorizontalPadding', () => {
    const { getByTestId } = render(<Screen noHorizontalPadding />);
    const AnimatedView = getByTestId(TestIDs.tests.Screen.AnimatedView);

    expect(AnimatedView).toHaveStyle({ paddingHorizontal: 0 });
  });
});
