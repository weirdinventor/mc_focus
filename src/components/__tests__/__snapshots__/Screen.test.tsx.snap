// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS]: Screen renders correctly 1`] = `
<View
  style={
    [
      {
        "flex": 1,
        "height": "100%",
        "paddingVertical": 8,
        "width": "100%",
      },
      {
        "backgroundColor": undefined,
      },
      {
        "paddingHorizontal": 24,
      },
    ]
  }
  testID="SCREEN_ANIMATED_VIEW"
>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "off",
        "right": "off",
        "top": "additive",
      }
    }
    style={
      {
        "flex": 1,
      }
    }
    testID="SCREEN_SAFE_AREA_VIEW"
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flex": 1,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          [
            {
              "flexGrow": 1,
              "paddingVertical": 16,
            },
            undefined,
          ]
        }
        onContentSizeChange={[MockFunction]}
        onLayout={[MockFunction]}
        scrollEnabled={true}
        showsVerticalScrollIndicator={false}
        style={
          {
            "flex": 1,
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <View />
      </RCTScrollView>
    </View>
  </RNCSafeAreaView>
</View>
`;
